[Bug] [Workflow Instance] Setting any startup parameters for the workflow run, and an error occurs when opening the workflow instance

### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.


### What happened

<img width="1167" height="299" alt="Image" src="https://github.com/user-attachments/assets/b937d7a8-b2fb-4e11-b514-84b48cd896a4" />

Setting any startup parameters for the workflow run, and an error occurs when opening the workflow instance

### What you expected to happen

Clicking on the workflow instance does not report an error

### How to reproduce

1

### Anything else

_No response_

### Version

3.3.1

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

I’ve also found another way to reproduce it.

1. First, create a workflow, such as the follow shell task

<img width="511" height="816" alt="Image" src="https://github.com/user-attachments/assets/d9bebd25-cfc3-49e9-b423-9f037b76079d" />

2. Then start the task. Note that the startup parameters need to be configured properly when starting the task

<img width="520" height="823" alt="Image" src="https://github.com/user-attachments/assets/07dc7812-9504-4626-86a3-5a6097f6dee9" />

**At the same time, a record will be inserted into the t_ds_workflow_instance table as follows**

<img width="1183" height="379" alt="Image" src="https://github.com/user-attachments/assets/c96a822d-6fdd-476b-93ec-26d1b1d17716" />

3. Then modify the workflow instance, For example, I just changed the number printed by echo from 4 to 5

<img width="495" height="825" alt="Image" src="https://github.com/user-attachments/assets/218156d5-610c-46d3-8ecb-395d2a44cc46" />

4. Finally, save this instance, then you will see the following error message
```
Caused by: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize value of type `java.lang.String` from Array value (token `JsonToken.START_ARRAY`)
 at [Source: (String)"{"commandType":"START_PROCESS","subWorkflowInstance":false,"startNodes":[],"commandParams":[{"prop":"dt_yesterday","direct":"IN","type":"VARCHAR","value":"${system.biz.date}"}],"timeZone":"Asia/Shanghai"}"; line: 1, column: 93] (through reference chain: java.util.LinkedHashMap["commandParams"]) 
```

<img width="960" height="381" alt="Image" src="https://github.com/user-attachments/assets/2db14740-bfcb-44b2-afba-ee82df2faa1b" />

## RC

Combining the command_param metadata in `t_ds_workflow_instance` above and the code for updating workflow instance(`org.apache.dolphinscheduler.api.service.impl.WorkflowInstanceServiceImpl#updateWorkflowInstance`), the RC for the error can be found as follows:

When updating the workflow instance, the commandParamMap String is extracted from the metadata and parse it as an object of type <Map<String, String>. **However, in commandParamMap, the value of commandParams is an array type, so an error is bound to occur here**

<img width="1436" height="583" alt="Image" src="https://github.com/user-attachments/assets/2c1c0dad-ded6-4195-81d9-bf9a8edf659a" />

The same problem also exists in the interface for viewing variables

https://github.com/apache/dolphinscheduler/blob/b3a40d882f28bbaa2bae3755d6203c2ff0048cc0/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkflowInstanceServiceImpl.java#L667-L673

--- a/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkflowInstanceServiceImpl.java
+++ b/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkflowInstanceServiceImpl.java
@@ -25,6 +25,7 @@ import static org.apache.dolphinscheduler.common.constants.Constants.GLOBAL_PARA
 import static org.apache.dolphinscheduler.common.constants.Constants.LOCAL_PARAMS;
 import static org.apache.dolphinscheduler.common.constants.Constants.TASK_LIST;
 import static org.apache.dolphinscheduler.common.constants.Constants.WORKFLOW_INSTANCE_STATE;
+import static org.apache.dolphinscheduler.common.utils.JSONUtils.parseObject;
 import static org.apache.dolphinscheduler.plugin.task.api.TaskPluginManager.checkTaskParameters;
 
 import org.apache.dolphinscheduler.api.constants.ApiFuncIdentificationConstant;
@@ -79,6 +80,7 @@ import org.apache.dolphinscheduler.dao.repository.TaskInstanceDao;
 import org.apache.dolphinscheduler.dao.repository.WorkflowInstanceDao;
 import org.apache.dolphinscheduler.dao.repository.WorkflowInstanceMapDao;
 import org.apache.dolphinscheduler.dao.utils.WorkflowUtils;
+import org.apache.dolphinscheduler.extract.master.command.ICommandParam;
 import org.apache.dolphinscheduler.plugin.task.api.model.Property;
 import org.apache.dolphinscheduler.plugin.task.api.utils.ParameterUtils;
 import org.apache.dolphinscheduler.plugin.task.api.utils.TaskTypeUtils;
@@ -663,13 +665,12 @@ public class WorkflowInstanceServiceImpl extends BaseServiceImpl implements Work
             return result;
         }
 
-        //
-        Map<String, String> commandParamMap = JSONUtils.toMap(workflowInstance.getCommandParam());
-        String timezoneId = null;
-        if (commandParamMap == null || StringUtils.isBlank(commandParamMap.get(Constants.SCHEDULE_TIMEZONE))) {
+        String timezoneId;
+        final ICommandParam commandParam = parseObject(workflowInstance.getCommandParam(), ICommandParam.class);
+        if (commandParam == null || StringUtils.isBlank(commandParam.getTimeZone())) {
             timezoneId = loginUser.getTimeZone();
         } else {
-            timezoneId = commandParamMap.get(Constants.SCHEDULE_TIMEZONE);
+            timezoneId = commandParam.getTimeZone();
         }
 
         setWorkflowInstance(workflowInstance, scheduleTime, globalParams, timeout, timezoneId);
@@ -873,11 +874,12 @@ public class WorkflowInstanceServiceImpl extends BaseServiceImpl implements Work
             return result;
         }
 
-        Map<String, String> commandParam = JSONUtils.toMap(workflowInstance.getCommandParam());
         String timezone = null;
-        if (commandParam != null) {
-            timezone = commandParam.get(Constants.SCHEDULE_TIMEZONE);
+        final ICommandParam commandParam = parseObject(workflowInstance.getCommandParam(), ICommandParam.class);
+        if (commandParam != null && StringUtils.isBlank(commandParam.getTimeZone())) {
+            timezone = commandParam.getTimeZone();
         }
+
         Map<String, String> timeParams = BusinessTimeUtils
                 .getBusinessTime(workflowInstance.getCmdTypeIfComplement(),
                         workflowInstance.getScheduleTime(), timezone);

--- a/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkflowInstanceServiceImpl.java
+++ b/dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkflowInstanceServiceImpl.java
@@ -667,7 +667,7 @@ public class WorkflowInstanceServiceImpl extends BaseServiceImpl implements Work
 
         String timezoneId;
         final ICommandParam commandParam = parseObject(workflowInstance.getCommandParam(), ICommandParam.class);
-        if (commandParam == null || StringUtils.isBlank(commandParam.getTimeZone())) {
+        if (commandParam == null || StringUtils.isEmpty(commandParam.getTimeZone())) {
             timezoneId = loginUser.getTimeZone();
         } else {
             timezoneId = commandParam.getTimeZone();
@@ -876,7 +876,7 @@ public class WorkflowInstanceServiceImpl extends BaseServiceImpl implements Work
 
         String timezone = null;
         final ICommandParam commandParam = parseObject(workflowInstance.getCommandParam(), ICommandParam.class);
-        if (commandParam != null && StringUtils.isBlank(commandParam.getTimeZone())) {
+        if (commandParam != null && StringUtils.isNotEmpty(commandParam.getTimeZone())) {
             timezone = commandParam.getTimeZone();
         }
 

--- a/dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/service/WorkflowInstanceServiceTest.java
+++ b/dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/service/WorkflowInstanceServiceTest.java
@@ -647,6 +647,21 @@ public class WorkflowInstanceServiceTest {
                             taskRelationJson, taskDefinitionJson, "2020-02-21 00:00:00", true, "", "", 0);
             Assertions.assertEquals(Status.SUCCESS, processInstanceFinishRes.get(Constants.STATUS));
 
+            final RunWorkflowCommandParam runWorkflowCommandParam = RunWorkflowCommandParam.builder()
+                    .commandParams(Lists.newArrayList(Property.builder()
+                            .prop("name")
+                            .direct(Direct.IN)
+                            .type(DataType.VARCHAR)
+                            .value("commandParam")
+                            .build()))
+                    .timeZone("Asia/Shanghai")
+                    .build();
+            workflowInstance.setCommandParam(JSONUtils.toJsonString(runWorkflowCommandParam));
+            Map<String, Object> processInstanceFinishRes2 =
+                    workflowInstanceService.updateWorkflowInstance(loginUser, projectCode, 1,
+                            taskRelationJson, taskDefinitionJson, "2020-02-21 00:00:00", true, "", "", 0);
+            Assertions.assertEquals(Status.SUCCESS, processInstanceFinishRes2.get(Constants.STATUS));
+
             // success
             when(workflowDefinitionMapper.queryByCode(46L)).thenReturn(workflowDefinition);
             putMsg(result, Status.SUCCESS, projectCode);
@@ -811,15 +826,31 @@ public class WorkflowInstanceServiceTest {
                         .type(DataType.VARCHAR)
                         .value("commandParam")
                         .build()))
+                .timeZone("Asia/Shanghai")
                 .build();
         WorkflowInstance workflowInstance = getProcessInstance();
         workflowInstance.setCommandType(CommandType.SCHEDULER);
         workflowInstance.setScheduleTime(new Date());
-        workflowInstance.setGlobalParams("");
         workflowInstance.setCommandParam(JSONUtils.toJsonString(runWorkflowCommandParam));
+
         when(workflowInstanceMapper.queryDetailById(1)).thenReturn(workflowInstance);
         Map<String, Object> successRes = workflowInstanceService.viewVariables(1L, 1);
         Assertions.assertEquals(Status.SUCCESS, successRes.get(Constants.STATUS));
+
+        final RunWorkflowCommandParam commandParamWithEmptyTimeZone = RunWorkflowCommandParam.builder()
+                .commandParams(Lists.newArrayList(Property.builder()
+                        .prop("name")
+                        .direct(Direct.IN)
+                        .type(DataType.VARCHAR)
+                        .value("commandParam")
+                        .build()))
+                .build();
+        workflowInstance.setCommandType(CommandType.SCHEDULER);
+        workflowInstance.setScheduleTime(new Date());
+        workflowInstance.setCommandParam(JSONUtils.toJsonString(commandParamWithEmptyTimeZone));
+        Map<String, Object> successRes2 = workflowInstanceService.viewVariables(1L, 1);
+        Assertions.assertEquals(Status.SUCCESS, successRes2.get(Constants.STATUS));
+
     }
 
     /**

--- a/dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/service/WorkflowInstanceServiceTest.java
+++ b/dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/service/WorkflowInstanceServiceTest.java
@@ -67,9 +67,13 @@ import org.apache.dolphinscheduler.dao.repository.TaskInstanceContextDao;
 import org.apache.dolphinscheduler.dao.repository.TaskInstanceDao;
 import org.apache.dolphinscheduler.dao.repository.WorkflowInstanceDao;
 import org.apache.dolphinscheduler.dao.repository.WorkflowInstanceMapDao;
+import org.apache.dolphinscheduler.extract.master.command.RunWorkflowCommandParam;
 import org.apache.dolphinscheduler.plugin.task.api.TaskPluginManager;
+import org.apache.dolphinscheduler.plugin.task.api.enums.DataType;
 import org.apache.dolphinscheduler.plugin.task.api.enums.DependResult;
+import org.apache.dolphinscheduler.plugin.task.api.enums.Direct;
 import org.apache.dolphinscheduler.plugin.task.api.enums.TaskExecutionStatus;
+import org.apache.dolphinscheduler.plugin.task.api.model.Property;
 import org.apache.dolphinscheduler.service.expand.CuringParamsService;
 import org.apache.dolphinscheduler.service.model.TaskNode;
 import org.apache.dolphinscheduler.service.process.ProcessService;
@@ -796,6 +800,26 @@ public class WorkflowInstanceServiceTest {
         when(workflowInstanceMapper.queryDetailById(1)).thenReturn(null);
         Map<String, Object> processNotExist = workflowInstanceService.viewVariables(1L, 1);
         Assertions.assertEquals(Status.WORKFLOW_INSTANCE_NOT_EXIST, processNotExist.get(Constants.STATUS));
+    }
+
+    @Test
+    public void testViewVariablesWithStartingParam() {
+        final RunWorkflowCommandParam runWorkflowCommandParam = RunWorkflowCommandParam.builder()
+                .commandParams(Lists.newArrayList(Property.builder()
+                        .prop("name")
+                        .direct(Direct.IN)
+                        .type(DataType.VARCHAR)
+                        .value("commandParam")
+                        .build()))
+                .build();
+        WorkflowInstance workflowInstance = getProcessInstance();
+        workflowInstance.setCommandType(CommandType.SCHEDULER);
+        workflowInstance.setScheduleTime(new Date());
+        workflowInstance.setGlobalParams("");
+        workflowInstance.setCommandParam(JSONUtils.toJsonString(runWorkflowCommandParam));
+        when(workflowInstanceMapper.queryDetailById(1)).thenReturn(workflowInstance);
+        Map<String, Object> successRes = workflowInstanceService.viewVariables(1L, 1);
+        Assertions.assertEquals(Status.SUCCESS, successRes.get(Constants.STATUS));
     }
 
     /**

